### ISSUE TITLE
Support for additional methods

### ISSUE THREAD

**Issue by dave-b-uk** [dtcookie/dynatrace-spring-reactive-weblient-tagging#3]
Currently this library works fine for `.then()` 
Is there any plan to have `.flatMap` and `.map` supported as well? Thank you

**Comment by Dynatrace-Reinhard-Pilz** [comment]
You're right, there are a couple of use cases where that solution is unfortunately not complete enough - simply because quite a lot of methods of class `reactor.core.publisher.Mono` are defined with the modifier `final`.
In order to get around that a completely different approach would be necessary (essentially byte code modification on top of OneAgent). While that is technically possible, I'm hesitant to provide a solution like that at the moment. OneAgent sooner or later will contain out of the box support for the reactor core packages. This project is meant to provide a workaround until that happens.

Makes sense?

**Comment by dave-b-uk** [comment]
Makes perfectly sense :)
Do you reckon that OneAgent will provide such support anytime soon?

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Internal comments are as a matter of fact mentioning the method `.flatMap` and `tracing through ALL Mono Operations` - essentially exactly what you're hoping for.
It's difficult to say, I'm afraid, when it will be available. My expectation is that the topic will get dealt with by the end of the year. 
Implementation has not yet been scheduled in by product management, though. Therefore please don't take my prediction at this point as the ultimate truth.
You may want to upvote this [Product Idea](https://community.dynatrace.com/t5/Dynatrace-product-ideas/Support-for-Netty-Reactor-Core-Monoflux/idi-p/165859) in order to speed things up. Dynatrace PM pays attention to these community posts.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
@dave-b-uk would you be willing to test out a potential solution for me?
As we already had established, the methods `flatMap` and `.map` are not automatically producing end to end visibility with this solution. Reason for that is, that the decorator for `Mono` is not allowed to override these methods.
What you CAN do, however, is to explicitly decorate the resulting `Mono` objects, that are being produced when calling `flatMap` or `map'.

Example:
```
Dynatrace.decorate(Dynatrace.decorate(mono.flatMap(...)).flatMap(...))
```

thanks,
Reinhard
